oap-server/analyzer/meter-analyzer/src/main/java/org/apache/skywalking/oap/meter/analyzer/prometheus/rule/Rules.java

```diff
@@ -40,36 +49,76 @@
+                    rule = rule.trim();
+                    if (rule.startsWith("/")) {
+                        rule = rule.substring(1);
+                    }
+                    if (!rule.endsWith(".yaml") && !rule.endsWith(".yml")) {
+                        return rule + "{.yaml,.yml}";
                     }
-                    Rule rule = new Yaml().loadAs(r, Rule.class);
-                    rule.setName(fileName);
                     return rule;
-                } catch (IOException e) {
-                    LOG.debug("Reading file {} failed", f, e);
-                }
+                })
+                .collect(Collectors.toMap(rule -> rule, $ -> false));
+        List<Rule> rules;
+        try (Stream<Path> stream = Files.walk(root)) {
+            rules = stream
+                    .filter(it -> formedEnabledRules.keySet().stream()
+                                    .anyMatch(rule -> {
+                                        boolean matches = FileSystems.getDefault().getPathMatcher("glob:" + rule)
+                                                .matches(root.relativize(it));
+                                        if (matches) {
+                                            formedEnabledRules.put(rule, true);
+                                        }
+                                        return matches;
+                                    }))
+                    .map(pathPointer -> {
+                        Path relativize = root.relativize(pathPointer);
+                        String relativizePath = relativize.toString();
+                        relativizePath = relativizePath.substring(0, relativizePath.lastIndexOf("."));
+                        return getRulesFromFile(relativizePath, pathPointer);
+                    })
+                    .filter(Objects::nonNull)
+                    .collect(Collectors.toList()) ;
+        }
+
+        if (formedEnabledRules.containsValue(false)) {
+            List<String> rulesNotFound = formedEnabledRules.keySet().stream()
+                    .filter(rule -> !formedEnabledRules.get(rule))
+                    .collect(Collectors.toList());
+            throw new UnexpectedException("Some configuration files of enabled rules are not found, enabled rules: " + rulesNotFound);
+        }
+        return rules;
+    }
+
+    private static Rule getRulesFromFile(String ruleName, Path path) {
+        File file = path.toFile();
+        if (!file.isFile()) {
+            return null;
+        }
+        try (Reader r = new FileReader(file)) {
+            String fileName = file.getName();
+            if (fileName.startsWith(".")) {
                 return null;
-            })
-            .filter(Objects::nonNull)
-            .collect(Collectors.toList());
+            }
+            Rule rule = new Yaml().loadAs(r, Rule.class);
+            if (rule == null) {
+                return null;
+            }
+            rule.setName(ruleName);
```

Comment:
@yswdqz can you confirm whether rule.name is used anywhere? I locally search and find nowhere use rule.name. If you can confirm rule.name is never used, we may consider remove it

Reply:
For now it is not be used.
But maybe we will use it in the future? Such as LAL.